PostHog/Replay/PostHogReplayIntegration.swift

```diff
@@ -274,6 +286,9 @@
         }
 
         @objc private func snapshot() {
+            // TODO: add debouncer with debouncerDelayMs to take into account how long it takes to execute the
+            // snapshot method
```

Comment:
since you cannot offload this from the main thread, a debouncer mechanism would be handy, can be separated to PR to get this out of the door since the timer calls `snapshot` every 500ms already instead of for every frame